edu.harvard.i2b2.ontology/src/core/edu/harvard/i2b2/ontology/dao/ConceptDao.java

```diff
@@ -205,7 +207,157 @@ else if (synonym.length() > 2)
 		return queryResult;
 	}
 
-
+    public List<ConceptType> findAllChildrenByParent(final GetAllChildrenDataMessage allChildrenMsg, ProjectType projectInfo, DBInfoType dbInfo) throws I2B2DAOException, I2B2Exception, JAXBUtilException {
+        final GetAllChildrenType allChildrenType = allChildrenMsg.getAllChildrenType();
+
+        // find return parameters
+        String parameters = DEFAULT;
+        if (allChildrenType.getType().equals("limited")) {
+            parameters = LIMITED;
+        } else if (allChildrenType.getType().equals("core")) {
+            parameters = CORE;
+        } else if (allChildrenType.getType().equals("all")) {
+            parameters = CORE + ALL;
+        }
+
+        if (allChildrenType.isBlob() == true) {
+            parameters = parameters + BLOB;
+        }
+
+        String metadataSchema = dbInfo.getDb_fullSchema();
+        setDataSource(dbInfo.getDb_dataSource());
+        if (projectInfo.getRole().isEmpty()) {
+            log.error("no role found for this user in project: " + projectInfo.getName());
+            I2B2Exception e = new I2B2Exception("No role found for user");
+            throw e;
+        }
+
+        //extract table code
+        String tableCd = StringUtil.getTableCd(allChildrenType.getParent());
+        String tableName = null;
+        String tableSql = "select distinct(c_table_name) from " + metadataSchema + "table_access where c_table_cd = ?";
```

Comment:
constrain to active only Ontologies by constraining "status_cd=Y"

Reply:
@reetametta Thanks.  I can add that to the query.  Just a fyi, none of the queries in the ConceptDao.java has that constraint.

Reply:
Follow up:  Leave **status_cd=Y** out.  Don't add status code.